### PR TITLE
Fixes #17 Add tests and coverage support

### LINKED ISSUE #17: Add tests support and test coverage
Before start filling up the API controllers we should be able to test those in-memory implementations using TDD, for that we need to set up a library, namely Pytest, so we can test such controller. Also, we would like to have covered at least 85% of all code source.

### PR DESCRIPTION
**Pull request by EliuX**
This PR adds **pytest** and **coverage**, in order to run tests and coverage reports respectively. It also adds a sample smoke_test.py to very that the tests are well installed by checking that the a reference to the app can be accessed. For better comprehension, more instructions were added to the README.md.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+import pytest
+
+from time_tracker_api import create_app
+
+@pytest.fixture(scope='session')
+def app():
+    app = create_app()
+    return app
```
Comment (fabidick22) [inline tests/conftest.py]: `return create_app()`
it's just a suggestion

Comment (EliuX) [reply tests/conftest.py]: Makes sense, I just know that here in the middle I will have to do adjustments.

Comment (fabidick22) [reply tests/conftest.py]: Ah OK, I understand :+1:

File: README.md [hunk 2]
```diff
@@ -50,7 +55,47 @@ automatically [pip](https://pip.pypa.io/en/stable/) as well.
...
+
+### Test
+We are using Pytest](https://docs.pytest.org/en/latest/index.html) for tests. The tests are located in the package 
+`tests` and use the [conventions for python test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery).
+
+To run the tests just execute:
+
+```
+python3 -m pytest -v
+```
```
Comment (enriquezrene) [inline README.md]: I am getting an error, any idea?
`$ python3 -m pytest -v
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.7/site-packages/pytest.py", line 89, in <module>
    raise SystemExit(pytest.main())
  File "/usr/local/lib/python3.7/site-packages/_pytest/config/__init__.py", line 61, in main
    config = _prepareconfig(args, plugins)
  File "/usr/local/lib/python3.7/site-packages/_pytest/config/__init__.py", line 182, in _prepareconfig
    config = get_config()
  File "/usr/local/lib/python3.7/site-packages/_pytest/config/__init__.py", line 156, in get_config
    pluginmanager.import_plugin(spec)
  File "/usr/local/lib/python3.7/site-packages/_pytest/config/__init__.py", line 530, in import_plugin
    __import__(importspec)
  File "/usr/local/lib/python3.7/site-packages/_pytest/tmpdir.py", line 25, in <module>
    class TempPathFactory(object):
  File "/usr/local/lib/python3.7/site-packages/_pytest/tmpdir.py", line 35, in TempPathFactory
    lambda p: Path(os.path.abspath(six.text_type(p)))
TypeError: attrib() got an unexpected keyword argument 'convert'`

File: tests/conftest.py (hunk 1 above)
Comment (EliuX) [reply tests/conftest.py]: But I applied your suggestion tho 👍

File: README.md (hunk 2 above)
Comment (EliuX) [reply README.md]: Are you running it in the virtual environment?

Comment (enriquezrene) [reply README.md]: yes,

`
(.venv) MacBook-Pro-de-Rene:time-tracker-api moe$ python3 -m pytest -v
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/pytest.py", line 89, in <module>
    raise SystemExit(pytest.main())
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 61, in main
    config = _prepareconfig(args, plugins)
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 182, in _prepareconfig
    config = get_config()
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 156, in get_config
    pluginmanager.import_plugin(spec)
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 530, in import_plugin
    __import__(importspec)
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/_pytest/tmpdir.py", line 25, in <module>
    class TempPathFactory(object):
  File "/Users/moe/Documents/IOET/ernst-young/time-tracker-api/.venv/lib/python3.7/site-packages/_pytest/tmpdir.py", line 35, in TempPathFactory
    lambda p: Path(os.path.abspath(six.text_type(p)))
TypeError: attrib() got an unexpected keyword argument 'convert'
(.venv) MacBook-Pro-de-Rene:time-tracker-api moe$
`

Comment (EliuX) [reply README.md]: the you have a problem with your **python3** or **pytest** installation.
image

Comment (enriquezrene) [reply README.md]: Do you have env vars that need to be configured?

Comment (enriquezrene) [reply README.md]: Just in case. I solved this issue following these instructions:
https://stackoverflow.com/a/58189684/2697734

Comment (EliuX) [reply README.md]: I have `attrs==18.2.0` among my dependencies, but this is included probably by `pytest`.

**Pull request merged**